### PR TITLE
Cache invalidation on updated content

### PR DESCRIPTION
**Pull request by dsamojlenko**
# Summary | Résumé

Adds a Cache invalidation plugin and related config to force refresh on updated content. 

When a page is updated, this plugin will send cache invalidation requests to CloudFront for the page itself, and any relevant index or category pages.

**Waiting for #442 to get merged so that the IAM user and policy are deployed to Production, then the credentials should be added to github secrets before merging this.** 

There is one minor unrelated change included, see note: https://github.com/cds-snc/gc-articles/pull/436/files#r779654926

### CODE REVIEW

File: infrastructure/terragrunt/aws/ecs/ecs.tf [hunk 1]
```diff
@@ -52,6 +52,9 @@ data "template_file" "wordpress_container_definition" {
     S3_UPLOADS_KEY               = aws_secretsmanager_secret_version.s3_uploads_key.arn
     S3_UPLOADS_SECRET            = aws_secretsmanager_secret_version.s3_uploads_secret.arn
     S3_UPLOADS_BUCKET_URL        = "https://${var.domain_name}"
+    C3_AWS_ACCESS_KEY_ID         = aws_secretsmanager_secret_version.c3_aws_access_key_id.arn
+    C3_AWS_SECRET_ACCESS_KEY     = aws_secretsmanager_secret_version.c3_aws_secret_access_key.arn
+    C3_DISTRIBUTION_ID           = var.c3_distribution_id
```
Comment (patheard) [inline infrastructure/terragrunt/aws/ecs/ecs.tf:57]: Does this need to be in the `container-definitions/wordpress.json.tmpl` ECS container def as well?

File: infrastructure/terragrunt/aws/ecs/inputs.tf [hunk 2]
```diff
@@ -88,6 +88,22 @@ variable "s3_uploads_secret" {
...
+}
+
+variable "c3_aws_secret_access_key" {
+  description = "Credentials for cache busting IAM user"
+  type        = string
+  sensitive   = true
+}
+
+variable "c3_distribution_id" {
+  type = string
```
Comment (patheard) [inline infrastructure/terragrunt/aws/ecs/inputs.tf:104]: nit: could you add a description for this var

File: infrastructure/terragrunt/env/staging/ecs/terragrunt.hcl [hunk 3]
```diff
@@ -80,6 +80,8 @@ inputs = {
   database_password_secret_arn     = dependency.database.outputs.database_password_secret_arn
   database_proxy_security_group_id = dependency.database.outputs.database_proxy_security_group_id
 
+  c3_distribution_id = dependency.load-balancer.outputs.cloudfront_distribution_id
```
Comment (patheard) [inline infrastructure/terragrunt/env/staging/ecs/terragrunt.hcl:83]: Can you add this to the load balancer's `mock_outputs` above.

**Review by patheard — CHANGES_REQUESTED** [review]
Few minor changes but overall looks good!

File: infrastructure/terragrunt/aws/ecs/ecs.tf (hunk 1 above)
Comment (dsamojlenko) [reply infrastructure/terragrunt/aws/ecs/ecs.tf:57]: yup

**Review by patheard — APPROVED** [review]
👍🏻

**Pull request merged**